core/common/src/test/java/alluxio/collections/IndexedSetTest.java

```diff
@@ -44,76 +45,87 @@ public long longValue() {
   }
 
   private IndexedSet<Pair> mSet;
-  private IndexedSet.FieldIndex<Pair> mIntIndex;
-  private IndexedSet.FieldIndex<Pair> mLongIndex;
+  private IndexedSet.NonUniqueFieldIndex<Pair> mNonUniqueIntIndex;
+  private IndexedSet.NonUniqueFieldIndex<Pair> mUniqueLongIndex;
 
   /**
    * Sets up the fields before running a test.
    */
   @Before
   public void before() {
-    mIntIndex = new IndexedSet.FieldIndex<Pair>() {
+    mNonUniqueIntIndex = new IndexedSet.NonUniqueFieldIndex<Pair>() {
+      @Override
       public Object getFieldValue(Pair o) {
         return o.intValue();
       }
     };
-    mLongIndex = new IndexedSet.FieldIndex<Pair>() {
+    mUniqueLongIndex = new IndexedSet.NonUniqueFieldIndex<Pair>() {
+      @Override
       public Object getFieldValue(Pair o) {
         return o.longValue();
       }
     };
-    mSet = new IndexedSet<>(mIntIndex, mLongIndex);
+    mSet = new IndexedSet<>(mNonUniqueIntIndex, mUniqueLongIndex);
+    long l = 0;
     for (int i = 0; i < 3; i++) {
-      for (long l = 0; l < 3; l++) {
-        mSet.add(new Pair(i, l));
+      for (int k = 0; k < 3; k++) {
+        mSet.add(new Pair(i, l++));
       }
     }
   }
 
   /**
-   * Tests the {@link IndexedSet#contains(IndexedSet.FieldIndex, Object)} method.
+   * Tests the {@link IndexedSet#contains(IndexedSet.NonUniqueFieldIndex, Object)} method.
    */
   @Test
-  public void containsTest() {
+  public void UniqueContainsTest() {
     for (int i = 0; i < 3; i++) {
-      Assert.assertTrue(mSet.contains(mIntIndex, i));
+      Assert.assertTrue(mSet.contains(mNonUniqueIntIndex, i));
     }
-    Assert.assertFalse(mSet.contains(mIntIndex, 4));
-    for (long l = 0; l < 3; l++) {
-      Assert.assertTrue(mSet.contains(mLongIndex, l));
+    Assert.assertFalse(mSet.contains(mNonUniqueIntIndex, 4));
+  }
+
+  /**
+   * Tests the {@link IndexedSet#contains(IndexedSet.UniqueFieldIndex, Object)} method.
+   */
+  @Test
+  public void NonUniqueContainsTest() {
+    for (long l = 0; l < 9; l++) {
+      Assert.assertTrue(mSet.contains(mUniqueLongIndex, l));
     }
-    Assert.assertFalse(mSet.contains(mLongIndex, 4L));
+    Assert.assertFalse(mSet.contains(mUniqueLongIndex, 9L));
   }
 
   /**
-   * Tests the {@link IndexedSet#getByField(IndexedSet.FieldIndex, Object)} method.
+   * Tests the {@link IndexedSet#getByField(IndexedSet.NonUniqueFieldIndex, Object)} method.
    */
   @Test
-  public void getTest() {
+  public void nonUniqueGetTest() {
     for (int i = 0; i < 3; i++) {
-      Set<Pair> set = mSet.getByField(mIntIndex, i);
+      Set<Pair> set = mSet.getByField(mNonUniqueIntIndex, i);
       Assert.assertEquals(3, set.size());
       List<Long> longs = new ArrayList<>(set.size());
       for (Pair o : set) {
         longs.add(o.longValue());
       }
       Collections.sort(longs);
       for (int j = 0; j < 3; j++) {
-        Assert.assertEquals(new Long(j), longs.get(j));
+        Assert.assertEquals(new Long(i * 3 + j), longs.get(j));
```

Comment:
Yes, because the pair is like 
{(0,0), (0,1), (0,2), 
 (1,3), (1,4), (1,5), 
 (2,6)....};
 Each value of int has three different mapping values of long. And each long value only has one corresponding int value.